subxt/src/blocks/extrinsic_types.rs

```diff
@@ -707,12 +727,24 @@ impl<'a> ExtrinsicSignedExtension<'a> {
         self.ty_id
     }
 
+    /// DecodedValueThunk representing the type of the extra of this signed extension.
+    fn decoded(&self) -> Result<DecodedValueThunk, Error> {
+        let decoded_value_thunk = DecodedValueThunk::decode_with_metadata(
+            &mut &self.bytes[..],
+            self.ty_id,
+            self.metadata,
+        )?;
+        Ok(decoded_value_thunk)
+    }
+
     /// Signed Extension as a [`scale_value::Value`]
     pub fn value(&self) -> Result<DecodedValue, Error> {
-        let value =
-            DecodedValue::decode_as_type(&mut &self.bytes[..], self.ty_id, self.metadata.types())?;
+        self.decoded()?.to_value()
+    }
 
-        Ok(value)
+    /// Decodes the `extra` bytes of this Signed Extension into a static type.
+    pub fn as_type<E: DecodeAsType>(&self) -> Result<E, Error> {
+        self.decoded()?.as_type::<E>().map_err(Into::into)
```

Comment:
Please re-apply the changes that weren't properly merged here:

https://github.com/paritytech/subxt/pull/1209/commits/64b5c90852bfca7dfa00c8dc80e2818f05075bdb#diff-68ce0867464cd9e4eb9e44abb6a5f2b227d8cd4d75f237d17b77e0d98166c3ebR712